assets/src/edit-story/components/panels/borderRadius/panel.js

```diff
@@ -0,0 +1,198 @@
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { Row, Toggle, Numeric } from '../../form';
+import { SimplePanel } from '../panel';
+import { useCommonObjectValue } from '../utils';
+import { BorderLockLine, Lock, Unlock } from '../../../icons';
+import { canMaskHaveBorder } from '../../../masks';
+
+const TOGGLE_WIDTH = 30;
+const DEFAULT_BORDER_RADIUS = {
+  topLeft: 0,
+  topRight: 0,
+  bottomRight: 0,
+  bottomLeft: 0,
+  locked: true,
+};
+
+const LockToggle = styled(Toggle)`
+  position: absolute;
+  top: calc(50% + 2px);
+  left: calc(50% - ${TOGGLE_WIDTH / 2}px);
+`;
+
+const LockToggleRow = styled(Row)`
+  margin: 0;
+`;
+
+const BoxedNumeric = styled(Numeric)`
+  padding: 6px 6px;
+  border-radius: 4px;
+`;
+
+const TOP_LEFT = 'top-left';
+const TOP_RIGHT = 'top-right';
+const BOTTOM_LEFT = 'bottom-left';
+const BOTTOM_RIGHT = 'bottom-right';
+
+const Icon = styled.div`
+  flex: 0 0 32px;
+  text-align: center;
+  padding-top: 6px;
+  ${({ corner }) => TOP_RIGHT === corner && `transform: scaleX(-1);`}
+  ${({ corner }) => BOTTOM_LEFT === corner && `transform: scaleY(-1);`}
+  ${({ corner }) => BOTTOM_RIGHT === corner && `transform: scale(-1, -1);`}
+  svg {
+    width: 24px;
+  }
+`;
+
+function BorderRadiusPanel({ selectedElements, pushUpdateForObject }) {
+  const borderRadius = useCommonObjectValue(
+    selectedElements,
+    'borderRadius',
+    DEFAULT_BORDER_RADIUS
+  );
+
+  const allSupportBorder = selectedElements.every((el) =>
+    canMaskHaveBorder(el)
+  );
+
+  const lockRadius = borderRadius.locked === true;
+
+  const handleChange = useCallback(
+    (name, value) => {
+      const newRadius = !lockRadius
+        ? {
+            [name]: value,
+          }
+        : {
+            topLeft: value,
+            topRight: value,
+            bottomRight: value,
+            bottomLeft: value,
+          };
+      pushUpdateForObject(
+        'borderRadius',
+        newRadius,
+        DEFAULT_BORDER_RADIUS,
+        true
+      );
+    },
+    [pushUpdateForObject, lockRadius]
+  );
+
+  const handleLockChange = useCallback(
+    (locked) => {
+      const newRadius = locked
+        ? {
+            locked,
+            topLeft: borderRadius.topLeft,
+            topRight: borderRadius.topLeft,
+            bottomRight: borderRadius.topLeft,
+            bottomLeft: borderRadius.topLeft,
+          }
+        : {
+            locked,
+          };
+      pushUpdateForObject(
+        'borderRadius',
+        newRadius,
+        DEFAULT_BORDER_RADIUS,
+        true
+      );
+    },
+    [pushUpdateForObject, borderRadius]
+  );
+
+  if (!allSupportBorder) {
+    return null;
+  }
+
+  return (
+    <SimplePanel name="borderRadius" title={__('Corner radius', 'web-stories')}>
+      <Row>
+        <BoxedNumeric
+          value={borderRadius.topLeft}
+          aria-label={__('Edit: Top left corner radius', 'web-stories')}
```

Comment:
We're getting rid of the `Edit:` prefixes, see #4604. Let's not add new ones in the meantime.

Reply:
Ah, good point, thanks!